match.go

```diff
@@ -9,7 +9,7 @@ import (
 // ErrBadPattern indicates a globbing pattern was malformed.
 var ErrBadPattern = path.ErrBadPattern
 
-// Unix separator
+// The SFTP protocol always uses slash as the path separator.
 const separator = "/"
```

Comment:
We might be able to even just remove this entirely? 🤔  The only remaining use is in `cleanGlobPath`.